server/api/src/rsGeneralAdmin.cpp

```diff
@@ -52,6 +53,55 @@ namespace
         return std::find_if(b, e, [](unsigned char _ch) { return ::isspace(_ch); }) != e;
     } // contains_whitespace
 
+     auto throw_if_downgrading_irods_service_account_rodsadmin(
+        RsComm& rsComm,
+        const std::string_view _option,
+        const std::string_view _user_name,
+        const std::string_view _new_user_type
+    ) -> void {
+         if ("rodsadmin" == _new_user_type || "type" != _option) {
+             return;
+         }
+
+        BytesBuf* bbuf = nullptr;
+        irods::at_scope_exit free_buf { [&bbuf] { std::free(bbuf); }  };
+
+        if (const auto ec = rsZoneReport(&rsComm, &bbuf); ec < 0 ) {
+            const auto msg = fmt::format("[{}:{}] - Error fetching Zone Report", __func__, __LINE__);
+            addRErrorMsg(&rsComm.rError, ec , msg);
+            THROW(ec, msg);
+        }
+
+        const auto buf = static_cast<char*> (bbuf->buf);
+        try {
+            const nlohmann::json& zone_report = nlohmann::json::parse(buf, buf + bbuf->len);
+            const auto& zones = zone_report.at("zones");
+
+            const auto * local_zone_name = getLocalZoneName();
+
+            for ( const auto& zone : zones ) {
+                const auto& icat_server_env = zone.at("icat_server").at("service_account_environment");
+
+                if ( icat_server_env.at("irods_zone_name").get_ref<const std::string&>() != local_zone_name ) {
+                    continue;
+                }
+
+                for ( const auto& server : zone.at("servers") ) {
+                    const auto& server_admin = server.at("service_account_environment").at("irods_user_name").get_ref<const std::string&>();
+                    if ( server_admin  == _user_name ) {
+                        const auto& host_of_target_user = server.at("host_system_information").at("hostname").get_ref<const std::string&>();
+                        const auto msg = fmt::format( "Cannot downgrade another rodsadmin ({}) running another server in this zone. On server: {}", _user_name, host_of_target_user );
```

Comment:
We've been using square brackets to hold values in error messages...

so...
"rodsadmin ({})" -> "rodsadmin [{}]"

and then... the later part could be...
"another server [{}] in this zone."